barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.hpp

```diff
@@ -46,15 +48,25 @@ template <typename Curve> class ShpleminiProver_ {
                                                                        has_zk);
         // Create opening claims for Libra masking univariates
         std::vector<OpeningClaim> libra_opening_claims;
-        size_t idx = 0;
-        for (auto [libra_univariate, libra_evaluation] : zip_view(libra_univariates, libra_evaluations)) {
-            OpeningClaim new_claim;
-            new_claim.polynomial = Polynomial(libra_univariate);
-            new_claim.opening_pair.challenge = multilinear_challenge[idx];
-            new_claim.opening_pair.evaluation = libra_evaluation;
-            libra_opening_claims.push_back(new_claim);
-            idx++;
+        OpeningClaim new_claim;
+
+        if (has_zk) {
+            static constexpr FF subgroup_generator = Curve::SUBGROUP_GENERATOR;
+            const auto gemini_r = opening_claims[0].opening_pair.challenge;
+
+            std::array<std::string, 4> libra_eval_labels = {
```

Comment:
Could use std::string_view which is constexpr and make this whole array constexpr.